pkg/chunkenc/symbols.go

```diff
@@ -0,0 +1,374 @@
+package chunkenc
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"hash"
+	"io"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/loki/pkg/util"
+)
+
+// symbol holds reference to a label name and value pair
+type symbol struct {
+	Name, Value uint32
+}
+
+type symbols []symbol
+
+// symbolizer holds a collection of label names and values and assign symbols to them.
+// symbols are actually index numbers assigned based on when the entry is seen for the first time.
+type symbolizer struct {
+	mtx            sync.RWMutex
+	symbolsMap     map[string]uint32
+	labels         []string
```

Comment:
I find this name a bit confusing. IIUC, this is an array with all the symbols. I think it should be named `symbols`

Reply:
No, it is a list of labels in string form. `symbols` are actually references by index to the deduped list of labels.